### ISSUE TITLE
`redefined-while-unused` (`F811`) - false positive on method with the same name as an unused import

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#24915]
## [Playground](https://play.ruff.rs/d24d7803-ab82-4945-90ef-485da3d8712c)

```py
from foo import bar # error: F401 (correct)

class Foo:
    def bar(self): ... # error: F811 (false positive)

def bar(): ... # error: F811 (correct)
```
```
Redefinition of unused `bar` from line 1: `bar` redefined here (F811) [Ln 4, Col 9]
Redefinition of unused `bar` from line 1: `bar` redefined here (F811) [Ln 6, Col 5]
```

`Foo.bar` is not shadowning the top-level `bar` so the error should not be reported

### Version

v0.15.12
